server.js

```diff
@@ -82,6 +82,25 @@ app
       res.send(queryParams);
     })
 
+    server.get('/queue/list',async (req,res) => {
+      let types = ['waiting', 'active', 'delayed', 'completed', 'failed'];
+      let data = {};
+      (() => {
+        var promises = []
+        types.map(type => {
+          promises.push(new Promise(async(resolve, reject)=> {
+            let pdl_queue = await config.getNewQueue('pdl-queue').getJobs(type)
```

Comment:
This approach might work when the number of jobs is less. If the jobs go above (say 1000), it will take forever to load. I want the jobs to lazy load. If I click on a particular tab (say "Completed"), the data should be loaded only then and set to cache as well. The data for jobs of all type should not get loaded by default.

Reply:
Also, the jobs of each library have to be fetched independently. Currently, the jobs of both libraries are fetched at the same time.